File: src/integrations/sncf/credentials.ts
// Lecture des identifiants SNCF Connect depuis le Gestionnaire d'identifiants
// Windows (jamais en clair ailleurs, jamais transmis en dehors de cette
// machine). L'ecriture n'existe volontairement pas ici : la plateforme
// n'accepte jamais d'identifiants via HTTP, ils sont enregistres en dehors
// d'elle via `node deployment/enregistrer-identifiants-sncf.mjs` (qui duplique
// le meme mecanisme de stockage, cf. son propre commentaire d'en-tete).
// Porte depuis l'ancienne plateforme Flask
// (agents/generateur_ics_sncf/scraping.py), qui utilisait keyring (Python) -
// meme principe ici via @napi-rs/keyring.

import { Entry } from "@napi-rs/keyring";

const SERVICE_NAME = "EJAH-IcsSncf";
const EMAIL_KEY = "sncf_email";

export function obtenirIdentifiants(): { email: string; motDePasse: string } | null {
  const email = new Entry(SERVICE_NAME, EMAIL_KEY).getPassword();
  if (!email) return null;
  const motDePasse = new Entry(SERVICE_NAME, email).getPassword();
  if (!motDePasse) return null;
  return { email, motDePasse };
}

export function identifiantsConfigures(): boolean {
  return obtenirIdentifiants() !== null;
}

File: python/agents/voyages/app/integrations/sncf/credentials.py
# Créé par Cédric HOUPE.
# Usage ou reproduction à l'identique interdit sans autorisation.
"""Stockage des identifiants SNCF Connect dans le Gestionnaire
d'identifiants Windows (jamais en clair ailleurs, jamais transmis en
dehors de cette machine, jamais journalisés).

Migré depuis l'ancienne plateforme Flask
(agents/generateur_ics_sncf/scraping.py) vers l'agent EJAH "voyages" -
même mécanisme (module `keyring`), nom de service dédié à ce nouvel agent
(les identifiants enregistrés par une éventuelle ancienne installation ne
sont pas repris automatiquement)."""
from __future__ import annotations

import keyring

SERVICE_NAME = "EJAH-Voyages"
EMAIL_KEY = "sncf_email"


def enregistrer_identifiants(email: str, mot_de_passe: str) -> None:
    """Enregistre l'email et le mot de passe SNCF Connect dans le
    Gestionnaire d'identifiants Windows.

    Args:
        email: Adresse email du compte SNCF Connect.
        mot_de_passe: Mot de passe associé.
    """
    keyring.set_password(SERVICE_NAME, EMAIL_KEY, email)
    keyring.set_password(SERVICE_NAME, email, mot_de_passe)


def obtenir_identifiants() -> tuple[str, str] | tuple[None, None]:
    """Lit l'email et le mot de passe SNCF Connect depuis le Gestionnaire d'identifiants.

    Returns:
        Un tuple (email, mot_de_passe), ou (None, None) si rien n'est enregistré.
    """
    email = keyring.get_password(SERVICE_NAME, EMAIL_KEY)
    if not email:
        return None, None
    mot_de_passe = keyring.get_password(SERVICE_NAME, email)
    if not mot_de_passe:
        return None, None
    return email, mot_de_passe


def identifiants_configures() -> bool:
    """Indique si des identifiants SNCF Connect sont déjà enregistrés.

    Returns:
        True si un couple email/mot de passe est présent dans le Gestionnaire
        d'identifiants Windows.
    """
    email, mot_de_passe = obtenir_identifiants()
    return bool(email and mot_de_passe)
